index.html

```diff
@@ -559,53 +444,59 @@ <h3>
-          <a data-link-for="WakeLock">request()</a>, run these additional
-          steps:
-        </p>
-        <ol class="algorithm" data-link-for="WakeLock">
-          <li>Set |lock:WakeLock|.{{[[requestPromise]]}} to `undefined`.
-          </li>
-          <li>Abort the associated <a>request()</a> algorithm.
-          </li>
-        </ol>
         <p data-tests="wakelock-api.https.html" data-link-for="WakeLock">
-          The <a>request()</a> method, when invoked, MUST run the following
-          steps:
+          The <a data-lt="request">request(|type:WakeLockType|,
+          |options:WakeLockRequestOptions|)</a> method, when invoked,
+          MUST run the following steps:
         </p>
         <ol class="algorithm">
-          <li>Let |options:WakeLockRequestOptions| be the first argument.
+          <li>Let |promise:Promise| be <a data-cite=
+          "promises-guide#a-new-promise">a new promise</a>.
           </li>
-          <li>Let |lock:WakeLock| be the <a>context object</a>.
+          <li>Let |document:Document| be the <a>responsible document</a> of the
+          <a>current settings object</a>.
           </li>
-          <li>Let |promise:Promise| be <a data-cite=
-            "promises-guide#a-new-promise">a new promise</a>.
+          <li>If the <a>current global object</a> is the
+          {{DedicatedWorkerGlobalScope}} object and its <a>owner set</a>
+          <a data-xref-for="list">is empty</a>, reject |promise| with a
+          "{{NotAllowedError}}" {{DOMException}} and return |promise|.
           </li>
-          <li>If |lock|.{{[[requestPromise]]}} is not `undefined`,
-            reject |promise| with a "{{InvalidStateError}}" {{DOMException}},
-            and run <a>abort the request steps</a>.
+          <li>If the <a>current global object</a> is the {{Window}} object and
+          the |document|'s [=Document/browsing context=] is `null`, reject
+          |promise| with a "{{NotAllowedError}}" {{DOMException}} and
+          return |promise|.
           </li>
-          <li>If |options|.|signal:AbortSignal|’s <a>aborted flag</a> is set, then
-          reject |promise| with an "{{AbortError}}" {{DOMException}},
-          and run <a>abort the request steps</a>.
+          <li data-tests="wakelock-disabled-by-feature-policy.https.sub.html">
+          If |document| is not <a>allowed to use</a> the <a>policy-controlled
+          feature</a> named "`wake-lock`", reject |promise| with a
+          "{{NotAllowedError}}" {{DOMException}} and return |promise|.
+          </li>
+          <li>If the <a>user agent</a> <a data-lt="deny wake lock">denies the
+          wake lock</a> of this |type| for |document|, reject |promise|
+          with a "{{NotAllowedError}}" {{DOMException}} and return |promise|.
           </li>
-          <li>If |options|.|signal| is not `null`, then [=AbortSignal/add=] to |options|.|signal|:
+          <li>If |options|.|signal:AbortSignal| member is present, then:
```

Comment:
I think this needs to come a bit later in the algorithm, otherwise we can have the following sequence of events:
1. `options.signal` is present and the `abort` flag is _not_ set.
1. "Run release a wake lock with `promise` and `type`" is added to `options.signal`'s algorithms.
1. The "obtain permission" steps result in "denied", `promise` is rejected. We never reach "acquire a wake lock with `promise` and `type`" step and `promise` is never added to `[[ActiveLocks]]`.
1. Script calls `myAbortController.abort()`
1. "Run release a wake lock", but `promise` has already been rejected and it's not part of `[[ActiveLocks]]`.

Reply:
`abort` can always happen after the promise was rejected (or resolved) so we need to make the "release wake lock" algorithm handle that - I just did